appdatabase/database.go

```diff
@@ -71,3 +72,30 @@ func EncryptDatabase(oldPath, newPath, password string) error {
 func ChangeDatabasePassword(path, password, newPassword string) error {
 	return sqlite.ChangeEncryptionKey(path, password, newPassword)
 }
+
+// GetDBFilename takes an instance of sql.DB and returns the filename of the "main" database
+func GetDBFilename(db *sql.DB) (string, error) {
+	var i, category, filename string
+	rows, err := db.Query("PRAGMA database_list;")
+	if err != nil {
+		return "", err
+	}
+
+	defer rows.Close()
+	for rows.Next() {
+		err = rows.Scan(&i, &category, &filename)
+		if err != nil {
+			return "", err
+		}
+
+		// The "main" database is the one we care about
+		if category == "main" {
+			return filename, nil
+		}
+	}
+	if err := rows.Err(); err != nil {
+		return "", nil
```

Comment:
Should the err be returned here?
```suggestion
		return "", err
```

Reply:
🤦, thank you nice catch